File: 890-lemonade-change/lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        five = 0
        ten =0
        twen =0

        my_list= list()

        for i in range(0,len(bills)):

            if bills[i] == 5:
                five +=1
                # my_list.append(5)
               
            elif bills[i] == 10:
                if five>=1:
                    ten += 1
                    five -= 1
                    # my_list.append(10)
                    # my_list.remove(5)
                else:
                    return False

            else: 
                if ten >= 1 and five>=1:
                    twen +=1
                    ten -=1
                    five -= 1
                    # my_list.append(20)
                    # my_list.remove(10)
                    # my_list.remove(5)
                elif five>=3:
                    twen += 1
                    five -= 3
                    # my_list.append(20)
                    # for i in range(3): my_list.remove(5)
                else:
                    return False  

        return True
        

File: 890-lemonade-change/lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {

        int five=0,ten=0,twenty=0;
        int n=bills.length;
        if( bills[0]==10 || bills[0]==15 ) return false;

        for ( int i=0;i<n;i++ ) { 

            if( bills[i]==5 ) five++;

            if( bills[i]==10 ) { 
                if( five>=1 ) { 
                    five--;
                    ten++;
                }
                else return false;
            }

            if( bills[i]==20 ) {
                if( ten>=1 && five>=1) { 
                    ten--;
                    five--;
                    twenty++;
                } else if ( five>=3) { 
                    five -=3;
                    twenty++;
                } else { return false;}
            }
        }

        return true;

    }
}